src/content/en/2021/caching.md

```diff
@@ -1,20 +1,488 @@
+
+{{ figure_markup(
+  image="invalid-date-formats.png",
+  caption="Percent of responses with invalid date formats.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=383751268&format=interactive",
+  sheets_gid="1680471251",
+  sql_file="invalid_last_modified_and_expires_and_date.sql"
+) }}
+
+Between 2020 and 2021, the percent using invalid `Date` improved by 0.5% but worsened for `Last-Modified` and `Expires` showing that it was related to how the date was set on caching.
+
+This shows us that automation of the `Date` header could benefit from further attention...?
+
+### `Vary`
+
+An essential step in caching a resource is understanding if it was previously cached. The browser typically uses the url as the cache key. At the same time, requests with similar urls but different `Accept-Encoding` could be cached incorrectly. That's why we use the `Vary` header to instruct the browser to add a value of one or more headers to the cache key.
+
+{{ figure_markup(
+  image="vary-directives.png",
+  caption="Usage of `Vary` directives.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1279247484&format=interactive",
+  height=436,
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+The most popular `Vary` header is `Accept-Encoding` with 90.3% usage, followed by `User-Agent` with 10.9%, `Origin` with 10.1%, and `Accept` with 4.8%.
+
+This shows a 1.5% decrease in use of `Accept-Encoding` from 2020.
+
+{{ figure_markup(
+  caption="Percent of mobile responses that set the `Vary` header.",
+  content="46.3%",
+  classes="big-number",
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+It's important to point out that only 46.25% of total requests audited use the `Vary` header, but when compared to 2020, we see an overall increase by 2.85%.
+
+{{ figure_markup(
+  caption="Percent of mobile responses with the `Vary` header that also set `Cache-Control`.",
+  content="83.4%",
+  classes="big-number",
+  sheets_gid="1033782866",
+  sql_file="vary_directives.sql"
+) }}
+
+Of the requests using the `Vary` header, 83.4% also have the `Cache-control`. This shows us a 2.1% improvement from 2020.
+
+## Setting cookies on cacheable responses
+
+In the 2020 Caching chapter, we were reminded to be aware of using `set-cookie` with cacheable responses because only 4.9% of responses used the `private` directive, putting a user's private data at risk of being accidentally served to a different user via a CDN.
```

Comment:
Might be helpful to re-iterate an explanation from 2020 here of what `set-cookie` is and why we need to be aware of using it